snapcraft/plugins/ruby.py

```diff
@@ -0,0 +1,144 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2017 James Beedy <jamesbeedy@gmail.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""The ruby plugin is useful for ruby based parts.
+The plugin uses gem to install dependencies from a `Gemfile`.
+This plugin uses the common plugin keywords as well as those for "sources".
+For more information check the 'plugins' topic for the former and the
+'sources' topic for the latter.
+Additionally, this plugin uses the following plugin-specific keywords:
+    - gems:
+      (list)
+      A list of gems to install.
+    - ruby-version:
+      (string)
+      The version of ruby you want this snap to run.
+"""
+import re
+
+import platform
+
+from os import makedirs, environ
+from os.path import exists, join
```

Comment:
Please just import "os" and use the functions with their namespace. I realize the snapcraft codebase isn't enitrely consistent here, but for new code it's easier to read if one doesn't need to double-check the imports in different files.